enzyme/Enzyme/AdjointGenerator.h

```diff
@@ -4932,6 +4931,35 @@ class AdjointGenerator
             }
           } else
             assert(0 && "illegal mpi");
+        } else if (Mode == DerivativeMode::ForwardMode) {
+          IRBuilder<> Builder2(&call);
+          getForwardBuilder(Builder2);
+
+          Value *buf = gutils->invertPointerM(call.getOperand(0), Builder2);
+          Value *count = gutils->getNewFromOriginal(call.getOperand(1));
+          Value *datatype = gutils->getNewFromOriginal(call.getOperand(2));
+          Value *source = gutils->getNewFromOriginal(call.getOperand(3));
+          Value *tag = gutils->getNewFromOriginal(call.getOperand(4));
+          Value *comm = gutils->getNewFromOriginal(call.getOperand(5));
+          Value *request = gutils->invertPointerM(call.getOperand(6), Builder2);
```

Comment:
Can you assert buf and request are active?

Reply:
done